### PR TITLE
Prefs fix

### PR DESCRIPTION
**Pull request by JasonLG1979**

### CODE REVIEW

File: pithos/pithos.py [hunk 1]
```diff
@@ -1051,24 +1054,80 @@ def show_about(self, version):
...
+        self.last_control_proxy_pac = self.prefs_dlg.control_proxy_pac_entry.get_text()
+        self.last_one_checkbox = self.prefs_dlg.pandora_one_checkbutton.get_active()
+        self.last_volume = self.settings.get_value('volume')
+
         self.settings.delay() # Dialog will apply
         self.prefs_dlg.show()
 
+    def on_password_entry_grab_focus(self, *ignore):
+        #clear out the password_entry and uncheck 
+        #pandora_one_checkbutton when it's clicked into.
```
Comment (TingPing) [inline pithos/pithos.py]: Uh.. why?

File: pithos/pithos.py [hunk 2]
```diff
@@ -1051,24 +1054,80 @@ def show_about(self, version):
...
+        new_control_proxy = self.prefs_dlg.control_proxy_entry.get_text()
+        new_control_proxy_pac = self.prefs_dlg.control_proxy_pac_entry.get_text()
+        new_one_checkbox = self.prefs_dlg.pandora_one_checkbutton.get_active()
 
-        email = self.settings['email']
         if response == Gtk.ResponseType.APPLY:
-            self.on_explicit_content_filter_checkbox()
-            if get_account_password(email) != self.last_pass:
+            if self.last_password != new_password or self.last_email != new_email:
+                #full stop, reset and reconnect on password and/or email change.
```
Comment (TingPing) [inline pithos/pithos.py]: The style for comments is generally one space after the comment and correct capitalization.

Comment (JasonLG1979) [inline pithos/pithos.py]: OK

File: pithos/pithos.py (hunk 1 above)
Comment (JasonLG1979) [inline pithos/pithos.py]: Why not? It's not uncommon for masked text entries to clear when you click into them...

Comment (TingPing) [inline pithos/pithos.py]: Think of all the elderly who will have a heart attack when they lose their password!

(but seriously I don't like it and have never seen such a thing in my life)

Comment (JasonLG1979) [inline pithos/pithos.py]: They wouldn't loose anything, click cancel. Whatever though no big deal I can get rid of it.

**Comment by JasonLG1979** [comment]
Better?

File: pithos/pithos.py [hunk 3]
```diff
@@ -1051,24 +1053,74 @@ def show_about(self, version):
...
+        self.last_proxy = self.prefs_dlg.proxy_entry.get_text()
+        self.last_control_proxy = self.prefs_dlg.control_proxy_entry.get_text()
+        self.last_control_proxy_pac = self.prefs_dlg.control_proxy_pac_entry.get_text()
+        self.last_one_checkbox = self.prefs_dlg.pandora_one_checkbutton.get_active()
+        self.last_volume = self.settings.get_value('volume')
+
         self.settings.delay() # Dialog will apply
         self.prefs_dlg.show()
 
+    def on_email_or_password_change(self, *ignore):
```
Comment (TingPing) [inline pithos/pithos.py]: I'm not sure this entire section makes sense, pandora is only reconnected after Apply.

Comment (JasonLG1979) [inline pithos/pithos.py]: You'll have to be more specific, what do you mean? on_email_or_password_change?

Comment (TingPing) [inline pithos/pithos.py]: Yes.

**Comment by TingPing** [comment]
@JasonLG1979 Let me know if I missed anything in my version.

**Pull request closed without merging**